ghga_transpiler/config/config.py

```diff
@@ -0,0 +1,94 @@
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Module to process config file"""
+
+from os.path import exists
+from pathlib import Path
+from typing import List, Optional
+
+import yaml
+from pydantic import BaseModel, root_validator
+
+from ..core.core import VERSION
+from .exceptions import DuplicatedName, MissingConfigFile
+
+HERE = Path(__file__).parent.resolve()
+CONFIG_LOCATION = HERE / "configs" / f"worksheet_config_{VERSION}.yaml"
+
+
+def read_config():
+    """Function to load yaml file"""
+
+    if exists(CONFIG_LOCATION):
+        with open(CONFIG_LOCATION, "r", encoding="utf-8") as file:
+            return yaml.safe_load(file)
+    raise MissingConfigFile(f"Config file for version {VERSION} cannot be found.")
+
+
+class DefaultSettings(BaseModel):
+    """A data model for the defaults of the per-worksheet settings of a transpiler config"""
+
+    start_row: int = 0
+    start_column: int = 0
+    end_column: int = 0
+
+
+class WorksheetSettings(BaseModel):
+    """A data model for the per-worksheet settings of a transpiler config"""
+
+    name: str
+    start_row: Optional[int]
+    start_column: Optional[int]
+    end_column: Optional[int]
+
+
+class Worksheet(BaseModel):
+    """A data model for worksheets in the transpiler config"""
+
+    sheet_name: Optional[str]
+    settings: Optional[WorksheetSettings]
+
+
+class Config(BaseModel):
+    """A data model for the transpiler config"""
+
+    ghga_version: Optional[str]
+    default_settings: DefaultSettings
+    worksheets: List[Worksheet]
+
+    @root_validator(pre=False)
+    def get_param(cls, values):  # pylint: disable=no-self-argument
+        """Function to manage parameters of global and worksheet specific configuration"""
+        for sheet in values.get("worksheets"):
+            for key in values.get("default_settings").__dict__:
+                if getattr(sheet.settings, key) is None:
+                    val = getattr(values.get("default_settings"), key)
+                    setattr(sheet.settings, key, val)
+        return values
+
+    @root_validator(pre=True)
+    def check_name(cls, values):  # pylint: disable=no-self-argument
+        """Function to manage parameters of global and worksheet specific configuration"""
+        names = []
+        for sheet in values.get("worksheets"):
+            if sheet["settings"]["name"] not in names:
+                names.append(sheet["settings"]["name"])
+            else:
+                raise DuplicatedName(
+                    "worksheet.settings name field must be unique in config"
+                )
```

Comment:
```suggestion
        # Check for duplicate attribute names
        attrs_counter = Counter(ws.settings.name for ws in values["worksheets"])
        dup_attrs = [ name for name, count in attrs_counter.items() if count > 1 ]
        if dup_attrs:
            raise DuplicatedName("Duplicate target attribute names: " + ", ".join(dup_attrs))

        # Check for duplicate worksheet names
        attrs_counter = Counter(ws.sheet_name for ws in values["worksheets"])
        dup_ws_names = [ name for name, count in attrs_counter.items() if count > 1 ]
        if dup_ws_names:
            raise DuplicatedName("Duplicate worksheet names: " + ", ".join(dup_ws_names))
```
We need to check the same thing for the worksheet names, right?  The suggested refactoring needs `from collections import Counter`.